lib/assert.js

```diff
@@ -285,15 +285,24 @@ function _deepEqual(actual, expected, strict, memos) {
   // Note: this accounts for both named and indexed properties on Arrays.
 
   // Use memos to handle cycles.
-  memos = memos || { actual: [], expected: [] };
-  const actualIndex = memos.actual.indexOf(actual);
-  if (actualIndex !== -1) {
-    if (actualIndex === memos.expected.indexOf(expected)) {
+  if (!memos) {
+    memos = {
+      actual: { map: new Map(), position: 0 },
```

Comment:
@joyeecheung That thought occurred to me and I couldn’t really find a difference … would there be one? Is that going to be more performant?